### DISCUSSION TITLE
Where is the source of the default Grafana Dashboards from?

### DISCUSSION THREAD

**Discussion by persunde** [prometheus-operator/kube-prometheus#951] [Q&A] [answered]
I am wondering where the dashboards that are by default a part of the Grafana instance when you install the kube-prometheus on a cluster come from.

Where do kube-prometheus find the dashboards? 
Why does kube-prometheus select these dashboards and not others, is there somewhere else where the selection of the default Kubernetes dashboards are located and kept up to date?

I see that the dashboards are compiled to this file: `./manifests/grafana-dashboardDefinitions.yaml`, but I was hoping to find the source where these dashboards are taken from and if possible the reason why these dashboards are selected. 
Who updates the dashboards, where and when?

After reading more carefully, are the default dashboards taken from here?
https://github.com/kubernetes-monitoring/kubernetes-mixin

**Comment by paulfantom** [comment, marked as answer]
> After reading more carefully, are the default dashboards taken from here?
> https://github.com/kubernetes-monitoring/kubernetes-mixin

Partially yes. kube-prometheus uses the concept of [monitoring-mixins](https://monitoring.mixins.dev/) to vendor alerts, recording rules, and dashboards coming from various places.
As for dashboards they are coming from following projects:
- https://github.com/kubernetes-monitoring/kubernetes-mixin (kubernetes dashboards)
- https://github.com/prometheus/node_exporter/tree/master/docs/node-mixin (node_exporter ones)
- https://github.com/prometheus/prometheus/tree/master/documentation/prometheus-mixin (prometheus dashboards)

Recently we refactored most of the codebase so it should be clearer which dashboards we use. You can see that inclusion in https://github.com/prometheus-operator/kube-prometheus/blob/master/jsonnet/kube-prometheus/main.libsonnet#L61

**Reply by persunde** [reply]
Thank you so much.
It would be nice if you could add this to the Wiki or the Readme so that this information is easy to find without having the read the code.
